File: benchmark/c/cpw/multiply_large_integers_under_large_modulo/multiply_large_integers_under_large_modulo.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



long long multiply_large_integers_under_large_modulo ( long long a, long long b, long long mod ) ;
long long multiply_large_integers_under_large_modulo ( long long a, long long b, long long mod ) {
  long long res = 0;
  a %= mod;
  while ( b ) {
    if ( b & 1 ) res = ( res + a ) % mod;
    a = ( 2 * a ) % mod;
    b >>= 1;
  }
  return res;
}

File: experimental data/RQ1/translation_result/fail/c/cpw/gemini/Restart/few-shot-True/cot-go/candidates-1/multiply_large_integers_under_large_modulo/multiply_large_integers_under_large_modulo.rs

fn multiply_large_integers_under_large_modulo(a: i64, b: i64, mod_: i64) -> i64 {
    let mut res = 0;
    let mut a = a % mod_;
    let mut b = b;
    while b != 0 {
        if b & 1 != 0 {
            res = (res + a) % mod_;
        }
        a = (2 * a) % mod_;
        b >>= 1;
    }
    res
}
